src/irmin-pack/unix/pack_store.ml

```diff
@@ -94,7 +94,7 @@ struct
     let indexing_strategy = Conf.indexing_strategy config in
     let lru_size = Conf.lru_size config in
     let staging = Tbl.create 127 in
-    let lru = Lru.create lru_size in
+    let lru = Lru.create ~weight:Val.weight lru_size in
```

Comment:
I still think we need a new/updated way to configure the LRU. Could be `lru_mem_size` (like mentioned previously) or some new key that lets a user configure how to cap the LRU.

As it is, this code would behave strangely for existing configurations. Imagine somebody has `lru_size=5` to keep only 5 entries. That would not translate well to the change in this PR.

Reply:
Hum there's definitely something I don't get I think - as the current code is not supposed to change anything unless the size of blobs is large.

Reply:
Hmm, maybe it's me but I'll talk it out to find understanding.

Currently, `lru_size` is used to cap the number of entries in the LRU. Lets use the current default of `100_000` for consideration. With the change in this PR, I think this would translate to a max of ~100 kB of data in the LRU, which I am assuming is much less than what `100_000` entries would be in size.

Reply:
- for commits and nodes/inodes there is no changes compare to now
- for  blobs: their weight is normalised to size_of(v)/1kb - so you have blobs of less than a 1kb there shouldn't be any difference; if not, you'll have a bit less blobs. And we are now skipping blobs bigger than lru_size / 10 * 1000 which I think is ok?

Reply:
Done in #2060

Reply:
For tezos, alls the blobs are smaller than 1 kb it seems: https://github.com/Ngoguey42/segment_hangzhou/blob/34e300b94e1dbf01ab3f04e7667bbef604ae21e4//tree_of_cycle_445.ipynb

Reply:
Thanks for digging that up. What I see in that data:

- the vast majority of objects are < 512 bytes
- for >= 512, their average size is `19_500_000 / 11_826 = 1649` bytes (so these are the ones that will impact the entry number configuration)
- a true memory-based cap could allow more objects into the LRU since most blobs are very small